src/coreclr/gc/gc.cpp

```diff
@@ -11732,13 +11750,17 @@ HRESULT gc_heap::initialize_gc (size_t soh_segment_size,
         return E_FAIL;
     }
 #else //USE_REGIONS
-    bool separated_poh_p = use_large_pages_p && 
-                           heap_hard_limit_oh[soh] && 
-                           (GCConfig::GetGCHeapHardLimitPOH() == 0) && 
-                           (GCConfig::GetGCHeapHardLimitPOHPercent() == 0);
+    bool empty_poh_p = heap_hard_limit_oh[soh] && 
```

Comment:
not sure why we are changing this? you wanted to support specifying POH limit as 0 when the user is not using large pages? seems like something we should discuss first as it's a public surfacing change.